tests/testtransport.nim

```diff
@@ -0,0 +1,133 @@
+import unittest
+import chronos
+import ../libp2p/[connection, 
+                  transports/transport, 
+                  transports/tcptransport,
+                  multiaddress, 
+                  wire]
+
+suite "TCP transport":
+  test "test listener: handle write":
+    proc testListener(): Future[bool] {.async, gcsafe.} =
+      let ma: MultiAddress = Multiaddress.init("/ip4/0.0.0.0/tcp/0")
+      proc connHandler(conn: Connection): Future[void] {.async, gcsafe.} =
+        result = conn.write(cstring("Hello!"), 6)
+
+      let transport: TcpTransport = newTransport(TcpTransport)
+      asyncCheck await transport.listen(ma, connHandler)
+      let streamTransport: StreamTransport = await connect(transport.ma)
```

Comment:
You cannot connect to "0.0.0.0" on Windows, so you need to connect to "127.0.0.1" instead.

More details: https://github.com/golang/go/commit/1a0b1cc

Reply:
Welp...

Reply:
Very interesting undocumented feature of *nix BSD sockets, fixed in https://github.com/status-im/nim-chronos/commit/3b8874a9e89ba02c76c3e4db9c0018545d03c3f2